### ISSUE TITLE
If you cancel print (via USB) fans, bed and extruder still powered

### ISSUE THREAD

**Issue by tiwanacote** [daid/LegacyCura#1436]
I tried to use:

In printWindow.py I tried this but didn´t work. How could I send Gcode to power off devices when you cancel a print?

def OnCancel(self, e):
       self._printerConnection.sendCommand("M107")
       self._printerConnection.sendCommand("M190 S0")  
       self._printerConnection.sendCommand("M109 S0")
       self._printerConnection.cancelPrint()

Thankyou!

**Comment by tiwanacote** [comment]
Ok...   this is not an elegant way but works fine

in printWindow.py:

```
def OnCancel(self, e):  
    self._printerConnection.cancelPrint()
    self._printerConnection.openActiveConnection()
    self._printerConnection.sendCommand('M107')
    self._printerConnection.sendCommand('M190 S0')
    self._printerConnection.sendCommand('M109 S0')
    self._printerConnection.closeActiveConnection()
```

**Comment by Ghostkeeper** [comment]
What printer was this on?

**Comment by tiwanacote** [comment]
Sorry, good question...

I found it in Repraps setting machine (I left my printer powered all weekend long). May be end.gcode must be sent if "cancel print" button is pressed.

Thankyou

**Comment by Ghostkeeper** [comment]
Yeah, that might be necessary. Calling the ending g-code at cancel would be a good solution, I think. This should account also for printers that like to decide for themselves when to turn the fans (or the heaters) off after a print. It would be a better solution than to add M107 to that list.

EDIT: Removed my comments about the current behaviour. What I saw was for Cura version 2.1 but you are talking about 15.04.
